mongodb/src/mongodb/errors.bal

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Holds the details of an MongoDB error
+#
+# + message - Specific error message for the error
+# + cause - Cause of the error
+# + detail - Error detail
+public type ErrorDetail record {
+    string message;
+    error cause?;
+    string detail?;
```

Comment:
Is this used? Is this needed? Can't we manage with `message`?

Reply:
For server errors, we are returning all the details within cause and detail. The message is added in the  MongoDB ballerina client level. It may not be descriptive enough

Reply:
can't we move the message addition to the native method, and remove this field?

Reply:
We can.... but is it the standard in ballerina? Usually in java when wrapping exceptions throwable is never manipulated.